storage/range.go

```diff
@@ -1246,6 +1251,22 @@ func (r *Range) InternalMerge(batch engine.Engine, ms *engine.MVCCStats, args *p
 	reply.SetGoError(err)
 }
 
+// InternalTruncateLog discards a prefix of the raft log.
+func (r *Range) InternalTruncateLog(batch engine.Engine, ms *engine.MVCCStats, args *proto.InternalTruncateLogRequest, reply *proto.InternalTruncateLogResponse) {
+	kvs, err := engine.MVCCScan(batch,
```

Comment:
I think it might be worth doing something more efficient here:

    if _, err := engine.ClearRange(MVCCEncodeKey(start), MVCCEncodeKey(end)); err != nil {
      reply.SetGoError(err)
      return
    }